Makefile

```diff
@@ -31,22 +31,37 @@ CMAKE_TEST_VARIABLES += -DCMAKE_PROJECT_DIR=$(CMAKE_PROJECT_DIR)
 CMAKE_TEST_VARIABLES += -DSW_DIR=$(SW_DIR)
 CMAKE_TEST_VARIABLES += -DMODE=TESTING
 
+
+# Test directories
+TEST_DIRS := $(wildcard $(BUILD_TESTS_DIR)/software/*/*/*)
+
+# Find test programs
+TEST_PROGRAMS := $(foreach dir,$(TEST_DIRS),$(wildcard $(dir)/*_test))
+
+
 unit_test:
+	@make build_test
+	@make test 
+	@echo "--------- FINISH -------"
+
+build_test:
 ifeq ($(rebuild),1)
 	-@rm -rf $(BUILD_TESTS_DIR)
 endif
 	@mkdir -p $(BUILD_TESTS_DIR)
 	@echo "BUILD_DIR=$(BUILD_TESTS_DIR)"
 	cd $(BUILD_TESTS_DIR)/ && \
 	cmake -S $(CURDIR) -B $(BUILD_TESTS_DIR) $(CMAKE_TEST_VARIABLES)  ;\
-	make -j$(MAXIMUM_CPU_CORES)
-	(cd $(BUILD_TESTS_DIR) && \
-		ctest --verbose --output-on-failure --output-junit test_results.xml . && \
-		junit2html test_results.xml test_results.html)
-	mkdir -p $(UNIT_TESTS_INSTALL_DIR)
-	cp $(BUILD_TESTS_DIR)/test_results.xml $(UNIT_TESTS_INSTALL_DIR)/
-	cp $(BUILD_TESTS_DIR)/test_results.html $(UNIT_TESTS_INSTALL_DIR)/
+	make -j$(MAXIMUM_CPU_CORES) ;\
+
+test: $(TEST_PROGRAMS)
+	@echo "Running all test programs"
+	@for program in $^; do \
+		echo "-----------------------";\
+		echo "Running $$program"; \
+		$$program -v | sed -E 's#.*\.c:[0-9]+:TEST\(([^)]+)\)#TEST(\1)#g; s#/workspace/software/bsw/mcal/unity_swc/tests/##g'; \
+	done
 
```

Comment:
you removed the installation to CMake/output/tests 

We need test results reports uploaded to CI

Reply:
Added the installation process until it is replaced by gcovr reports.